packages/ui/src/design-system/pie-chart/pie-chart.component.tsx

```diff
@@ -15,20 +15,33 @@ import {
 } from './pie-chart.data';
 
 import type { ColorValueHex } from '../../types';
-import type { CellProps, TooltipProps } from 'recharts';
+import type { CellProps } from 'recharts';
 import type { PickByValue } from 'utility-types';
 
 type PieChartDataProps = Partial<{
   overrides: CellProps;
 }>;
 export type PieChartColor = ColorValueHex | PieChartGradientColor;
+
+interface TooltipContentRendererProps<T> {
+  active?: boolean;
+  payload?: { name?: string; payload?: T }[];
+}
+
+interface TooltipProps<T> {
+  content:
+    | React.ReactElement
+    | ((props: TooltipContentRendererProps<T>) => React.ReactNode);
+}
+
 interface PieChartBaseProps<T extends object> {
   animate?: boolean;
   colors?: PieChartColor[];
   data: (PieChartDataProps & T)[];
   direction?: 'clockwise' | 'counterclockwise';
-  tooltip?: TooltipProps<number, string>['content'];
+  tooltip?: TooltipProps<PieChartDataProps & T>['content'];
```

Comment:
I just tried and the intersection with `PieChartDataProps` should be redundant

```suggestion
  tooltip?: TooltipProps<T>['content'];
```